### DISCUSSION TITLE
Thoughts on using konva for future vector manipulation?

### DISCUSSION THREAD

**Discussion by hluedeke** [Pattern-Projector/pattern-projector#307] [Ideas]
Howdy! Me again! I've been (lightly) looking at the embroidery stuff, so I wanted to propose an idea for future PP updates.

We have talked about vector manipulation a few times to be able to "edit" PDFs on the screen. I don't know how realistic that actually is because it's a huge undertaking. But if we do plan to go that route, then I would like to introduce konva (and react-konva) as a potential library candidate for drawing/vector management. It introduces concepts such as layers, paths, text, etc. for 2d canvas display/drawing as well as bounding boxes for dragging and rotating.

Now, I'm not saying we use this, but it would be a helpful tool to speed up some of the advanced concepts of embroidery file placement (which I'm actively working on). It would also help organize/speed up vector manipulation, should we ever choose to implement it. 

The biggest downside is that we are incorporating another library, particularly one of large size (1.4MB unzipped). Will that whole package be bundled into webpack? Probably not, but it's still big. And at the moment, it's not really integrated in PP at all, just this side little embroidery project that may not even have merit at the end of the day.

The alternative is to just code things by hand, which is still ok - it's just slower. I figured I would gauge how serious we are in addressing vector manipulation, and if it's unlikely to happen, then just drop the idea. If we do, however, plan to implement vector manipulation and/or advanced canvas UI, then I will happily integrate konva into the embroidery-flavored project I have going on right now.

**Comment by courtneypattison** [comment]
Hey :) I think if you're excited about this package and it'll be helpful, then go for it but probably as a forked project (as I mentioned in #300). My main concern with vector manipulation is that some vector pdfs are so badly formatted! Like the vogue patterns with tiny line segments. I'm leaning more toward something like clipping paths around pieces for pattern tetris because it would work with any pattern. That feels pretty far off right now though! I've looked a bit into svgedit for tracing patterns but I'm not sure PP is the best place for that sort of tool. Something like Inkscape/Gimp seems better suited.

**Reply by hluedeke** [reply]
> Something like Inkscape/Gimp seems better suited.

Agree here. Inkscape is what I use for vector manipulation + pattern tetris, and I will probably always prefer it. With the project forking, this becomes a non-issue. I will close this discussion to keep things clean 👍
